reactiveops/rok8s-scripts
# Added new object BG_DEPLOYMENT to do Blue Green deployments

Loosely based on  https://www.ianlewis.org/en/bluegreen-deployments-kubernetes
Overview:
* Added `-latest` to deployment name in the template
* Added `version:` label to deployment
* Added `version:` selector to the service, and hpa template
* Process:
  * Replace the `-latest` with ${CI_SHA1} ala the image name and `apply`
  * Update replica count with HPA cound 
  * Wait for ready replicas to equal expected replicas (will wait forever)
  * Patch service with new selector
  * Update HPA with new deployment name
  * Delete all deployments that have the begin with the same name as the current deployment  
    * for example, if new deployment name is  `example-d99bd40efa51dd92a143f0ca8280cbcbedc24acb` it will delete all deployments that match `grep example-`

I consider this WIP and should undergo thorough review before being merged
I was branched from the pip package branch because it was the easiest way for me to install locally. It does not actually require that work and we can remove it if necessary.
